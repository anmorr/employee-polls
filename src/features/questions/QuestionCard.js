import { Box, Container } from "@mui/system";
import CssBaseline from '@mui/material/CssBaseline';
import Typography from '@mui/material/Typography';
import TextField from '@mui/material/TextField';
import { Button } from "@mui/material";
import Copyright from "../../shared/components/Copyright";
import { createTheme, ThemeProvider } from '@mui/material/styles';


export default function QuestionCard({ optionOneProps, optionTwoProps, submitHandler }) {
    
    const theme = createTheme();

    return (
        <ThemeProvider theme={theme}>
            <Container component="main" maxWidth="xs">
            <CssBaseline />
            <Box
            sx={{
                marginTop: 8,
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                width: 500,
                maxWidth: '100%',
            }}
            >
          <h1>Would You Rather</h1>
          <Typography component="h5" variant="h5">
            Create Your Own Poll
          </Typography>
                    <Box component="form" onSubmit={submitHandler} noValidate sx={{ mt: 1 }}>
                        <Typography component="h5" variant="h5" sx={{
                        textAlign: 'center'
                    }}>
            First Option
          </Typography>            
            <TextField
                margin="normal"
                required
                fullWidth
                name="optionOne"
                label="Option One"
                type="text"
                id="optionOne"
                {...optionOneProps}
                multiline
                rows={4}
                inputProps={{"data-testid": "option-one"}}
                        />
            <Typography component="h5" variant="h5" sx={{
                        textAlign: 'center'
                    }}>
            Second Option
          </Typography>         
            <TextField
                margin="normal"
                required
                fullWidth
                name="optionTwo"
                label="Option Two"
                type="text"
                id="optionTwo"
                {...optionTwoProps}
                multiline
                rows={4}
                inputProps={{"data-testid": "option-two"}}
                            
                />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2, }}
              disabled={
                optionOneProps.value &&
                optionTwoProps.value ?
                  false :
                  true
              }
                            data-testid="submit-button"
                            
            >
              Submit
            </Button>
          </Box>
        </Box>
        <Copyright sx={{ mt: 8, mb: 4 }} />
      </Container>
    </ThemeProvider>
    )
}