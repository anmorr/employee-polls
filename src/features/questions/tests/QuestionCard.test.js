import {
    render
} from '@testing-library/react';
import QuestionCard from '../QuestionCard';

const optionOneProps = {
    value: 'Go to LA',
    onChange: e => setValue(e.target.value)
}

const optionTwoProps = {
    value: 'Go to MIA',
    onChange: e => setValue(e.target.value)
}

const handleSubmit = () => {

}
        

describe('QuestionCard', () => {

    it('renders new question', () => {
        render(
            <QuestionCard
                optionOneProps={optionOneProps}
                optionTwoProps={optionTwoProps}
                submitHandler={handleSubmit} />
        );
    });

    it('matches the snapshot', () => {
        var component = (
            <QuestionCard
                optionOneProps={optionOneProps}
                optionTwoProps={optionTwoProps}
                submitHandler={handleSubmit} />
        );
        expect(component).toMatchSnapshot();
    });

})
