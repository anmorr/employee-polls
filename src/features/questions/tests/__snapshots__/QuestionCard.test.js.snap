// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard matches the snapshot 1`] = `
<QuestionCard
  optionOneProps={
    Object {
      "onChange": [Function],
      "value": "Go to LA",
    }
  }
  optionTwoProps={
    Object {
      "onChange": [Function],
      "value": "Go to MIA",
    }
  }
  submitHandler={[Function]}
/>
`;
